lib/foreman_maintain/concerns/firewall/maintenance_mode.rb

```diff
@@ -0,0 +1,37 @@
+module ForemanMaintain
+  module Concerns
+    module Firewall
+      module MaintenanceMode
+        def notify_and_ask_to_install_firewall_utility
+          puts 'Unable to find nftables or iptables!'
+          question, pkg = question_and_pkg_name
+          answer = ask_decision(question, actions_msg: 'y(yes), q(quit)')
+          if answer == :yes
+            packages_action(:install, pkg)
+            feature(:instance).firewall.enable_maintenance_mode
+          end
+        end
+
+        def can_install_nft?
+          nft_kernel_version = Gem::Version.new('3.13')
```

Comment:
I think instead if checking "would this kernel support nft", I'd go with "did that OS release come with nft by default". That'd be el 8+ and for Debian/Ubuntu I can look it up for you if you want.